### PR TITLE
Suppress unsupported transitive dependencies

### PR DESCRIPTION
**Pull request by sharwell**
- Revert #59324
- Revert #59230
- Supersedes #59341
- ~~Disable new NuGet check which is causing restore problems in internal previews~~ (removed this change since it was unnecessary)
- Suppress unsupported transitive dependencies prior to 17.2 public release

### CODE REVIEW

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj [hunk 1]
```diff
@@ -40,7 +40,12 @@
     <PackageReference Include="Microsoft.VisualStudio.ImageCatalog" Version="$(MicrosoftVisualStudioImageCatalogVersion)" PrivateAssets="all" />
     <PackageReference Include="Microsoft.VisualStudio.Language" Version="$(MicrosoftVisualStudioLanguageVersion)" />
     <!-- No warn on NU1701 until the LSP client targets netstandard2.0 https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1369985/ -->
-    <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Client" Version="$(MicrosoftVisualStudioLanguageServerClientVersion)" NoWarn="NU1701" />
+    <PackageReference Include="Microsoft.VisualStudio.LanguageServer.Client" Version="$(MicrosoftVisualStudioLanguageServerClientVersion)" PrivateAssets="all" NoWarn="NU1701" />
```
Comment (dibarbet) [inline src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:43]: @sharwell I think there is a version of the LSP client targeting .netstandard now (17.2.39)
I will checkout your branch and test it to see if it removes the warnings.

Comment (sharwell) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:43]: ➡️ The package may be available, but we can't adopt it prior to 17.2 public release.

**Review by JoeRobich — APPROVED** [review]

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj (hunk 1 above)
Comment (tmat) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:43]: FYI: All these will go away with my PR https://github.com/dotnet/roslyn/pull/59316

**Comment by jasonmalinowski** [comment]
Can somebody here summarize what's all going on here?

**Comment by sharwell** [comment]
> Can somebody here summarize what's all going on here?

Microsoft.VisualStudio.LanguageServer.Client version 17.0 is only compatible with net472. We reference the package here from a netcoreapp3.1 build and specify that NuGet should ignore NU1701 (no compatible framework available). NuGet changed the behavior to validate _transitive_ dependencies in the same way, so for each transitive dependency of Microsoft.VisualStudio.LanguageServer.Client that produced a NU1701 warning I did the following:

1. Add the package explicitly so I can set `NoWarn` to include NU1701
2. Declare the package as `PrivateAssets="all"` so downstream netcoreapp3.1 projects do not face the same problem
3. Declare the package as `ExcludeAssets="all"` so the contents of the package are completely excluded from all aspects of building this project

File: src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj (hunk 1 above)
Comment (tmat) [reply src/EditorFeatures/Core/Microsoft.CodeAnalysis.EditorFeatures.csproj:43]: Actually, i'm not changing EditorFeatures. I thought this is another project.

**Pull request merged**